src/wallet/asyncrpcoperation_common.cpp

```diff
@@ -40,3 +41,115 @@ std::pair<CTransaction, UniValue> SignSendRawTransaction(UniValue obj, std::opti
+                        "`AllowRevealedAmounts` or weaker if you wish to allow this transaction to "
+                        "proceed anyway.");
+                case AddressResolutionError::ChangeAddressSelectionError:
+                    throw JSONRPCError(
+                        RPC_INVALID_PARAMETER,
+                        "Could not select a change address that allows this transaction "
+                        "to proceed without publicly revealing transaction details. THIS MAY AFFECT "
+                        "YOUR PRIVACY. Resubmit with the `privacyPolicy` parameter set to "
+                        "`AllowRevealedAmounts` or weaker if you wish to allow this transaction to "
+                        "proceed anyway.");
+                default:
+                    assert(false);
+            }
+        },
+        [&](const InvalidFundsError& err) {
+            bool isFromUa = std::holds_alternative<libzcash::UnifiedAddress>(selector.GetPattern());
+            throw JSONRPCError(
+                RPC_INVALID_PARAMETER,
+                strprintf(
+                    "Insufficient funds: have %s, %s",
+                    FormatMoney(err.available),
+                    std::visit(match {
+                        [](const InsufficientFundsError& ife) {
+                            return strprintf("need %s", FormatMoney(ife.required));
+                        },
+                        [](const DustThresholdError& dte) {
+                            return strprintf(
+                                    "need %s more to avoid creating invalid change output %s (dust threshold is %s)",
+                                    FormatMoney(dte.dustThreshold - dte.changeAmount),
+                                    FormatMoney(dte.changeAmount),
+                                    FormatMoney(dte.dustThreshold));
+                        }
+                    },
+                    err.reason))
+                    + (selector.TransparentCoinbasePolicy() != TransparentCoinbasePolicy::Disallow
+                       ? "" :
+                       "; note that coinbase outputs will not be selected if you specify "
+                       "ANY_TADDR, any transparent recipients are included, or if the "
+                       "`privacyPolicy` parameter is not set to `AllowRevealedSenders` or weaker")
+                    + (!isFromUa || strategy.AllowLinkingAccountAddresses() ? "." :
+                       ". (This transaction may require selecting transparent coins that were sent "
+                       "to multiple Unified Addresses, which is not enabled by default because "
+                       "it would create a public link between the transparent receivers of these "
+                       "addresses. THIS MAY AFFECT YOUR PRIVACY. Resubmit with the `privacyPolicy` "
+                       "parameter set to `AllowLinkingAccountAddresses` or weaker if you wish to "
```

Comment:
This message will need to change to be less UA-specific if we make the change to the meaning of `AllowRevealedSenders`.

Reply:
See #6467.